plugin/src/main/java/org/opennms/velocloud/requisition/CustomerRequisitionProvider.java

```diff
@@ -69,19 +229,22 @@ public static class Request extends AbstractRequisitionProvider.Request {
         public Request() {}
 
         public Request(final Connection connection) {
-            super("velocloud-customer", connection);
-            this.enterpriseId = connection.getEnterpriseId().orElse(null);
+            this(connection, connection.getEnterpriseId());
         }
 
-        private UUID enterpriseId;
+        public Request(final Connection connection, Optional<UUID> enterpriseId) {
+            super(TYPE, connection);
+            this.enterpriseId = enterpriseId;
+        }
+
+        private Optional<UUID> enterpriseId;
```

Comment:
`Optional` is only meant to be a return value (or an intermediate variable). Reason for this is: with `Optional` in a field, there are two kinds of `null` value now - which makes things even worse.